### PR TITLE
Add tag selection screen to edit view

### PR DESCRIPTION
**Pull request by arturs105**
Ref #519 

This only adds the selection screen, changes are not yet persisted.

### CODE REVIEW

File: Toggl.Foundation.MvvmCross/ViewModels/SelectTagsViewModel.cs [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+            Ensure.Argument.IsNotNull(navigationService, nameof(navigationService));
+
+            this.dataSource = dataSource;
+            this.navigationService = navigationService;
+
+            CloseCommand = new MvxAsyncCommand(close);
+            SaveCommand = new MvxAsyncCommand(save);
+            SelectTagCommand = new MvxCommand<SelectableTag>(selectTag);
+        }
+
```
Comment (willsb) [inline Toggl.Foundation.MvvmCross/ViewModels/SelectTagsViewModel.cs]: Extra newline

File: Toggl.Foundation.MvvmCross/ViewModels/SelectTagsViewModel.cs [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+            tags.Select(createSelectableTag)
+                .GroupBy(tag => tag.Workspace)
+                .Select(grouping => new SelectableTagCollection(grouping.Key, grouping))
+                .ForEach(Tags.Add);
+        }
+
+        private SelectableTag createSelectableTag(TagSuggestion tagSuggestion)
+        {
+            return new SelectableTag(
```
Comment (willsb) [inline Toggl.Foundation.MvvmCross/ViewModels/SelectTagsViewModel.cs]: Could be expression bodied member

File: Toggl.Foundation.MvvmCross/ViewModels/SelectTagsViewModel.cs [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+                tagSuggestion,
+                selectedTagIds.Contains(tagSuggestion.TagId)
+            );
+        }
+
+        private Task close()
+            => navigationService.Close(this);
```
Comment (willsb) [inline Toggl.Foundation.MvvmCross/ViewModels/SelectTagsViewModel.cs]: Cache and return the default array here

File: Toggl.Foundation.MvvmCross/ViewModels/EditTimeEntryViewModel.cs [hunk 4]
```diff
@@ -207,5 +210,7 @@ private Task close()
             StartTime = selectedDuration.Start;
             StopTime = selectedDuration.Stop;
         }
+
+        private Task selectTags() => navigationService.Navigate<SelectTagsViewModel>();
```
Comment (willsb) [inline Toggl.Foundation.MvvmCross/ViewModels/EditTimeEntryViewModel.cs]: I figured you're still not consuming the tags here and this will be done in a different PR, correct?

Comment (arturs105) [reply Toggl.Foundation.MvvmCross/ViewModels/EditTimeEntryViewModel.cs]: Exactly
